packages/jest-cli/src/runTest.js

```diff
@@ -19,11 +19,34 @@ const {
   NullConsole,
   setGlobal,
 } = require('jest-util');
+
+const fs = require('fs');
+const docblock = require('jest-docblock');
+const {getTestEnvironment} = require('jest-config');
 const getConsoleOutput = require('./reporters/getConsoleOutput');
 
 function runTest(path: Path, config: Config, resolver: Resolver) {
+  let data;
+
+  try {
+    data = fs.readFileSync(path, 'utf8');
+  } catch (e) {
+    return Promise.reject(e);
+  }
+  let testEnvironment = config.testEnvironment;
+
+  const testEnvironmentDocblock = docblock.parse(docblock.extract(data));
```

Comment:
could we just do:

`const customEnvironment = docblock.parse(docblock.extract(data))['jest-environment']`

and then re-use that? We only need that one prop :)